src/zope/app/apidoc/tests.py

```diff
@@ -13,83 +13,353 @@
+    codemodule._cleanUp()
+
+def _tearDown_LayerPlace(test):
+    _tearDown_AppSetup()
+
+
+class BrowserTestCase(unittest.TestCase):
+
+    layer = APIDocLayer
+
+    def setUp(self):
+        super(BrowserTestCase, self).setUp()
+        _setUp_AppSetup()
+        self._testapp = TestApp(self.layer.make_wsgi_app())
+
+    def tearDown(self):
+        _tearDown_AppSetup()
+        super(BrowserTestCase, self).tearDown()
+
+
+    def checkForBrokenLinks(self, orig_response, path,
+                            basic=None, max_links=None):
+        response = self.publish(path, basic=basic)
+        links = response.html.find_all('a')
+
+        seen = set()
+        for link in links:
+            try:
+                href = link.attrs['href']
+            except KeyError:
+                pass
+
+            if href.startswith('./'):
+                href = href[2:]
+
+            if href.startswith('http://localhost/'):
+                href = href[16:]
+
+            if not href.startswith('/'):
+                href = path.rsplit('/', 1)[0] + '/' + href
+
+            if href in seen:
+                continue
+            seen.add(href)
+
+            self.publish(href, basic=basic)
+
+            if max_links is not None:
+                max_links -= 1
+                if not max_links:
+                    break
+
+    def publish(self, path, basic=None, form=None, headers=None, env=None):
+        assert basic
+        self._testapp.authorization = ('Basic', tuple(basic.split(':')))
+        env = env or {}
+        env['wsgi.handleErrors'] = False
+        if form:
+            response = self._testapp.post(path, params=form,
+                                          extra_environ=env, headers=headers)
+        else:
+            response = self._testapp.get(path, extra_environ=env, headers=headers)
+
+        response.getBody = lambda: response.unicode_normal_body
+        response.getStatus = lambda: response.status_int
+        response.getHeader = lambda n: response.headers[n]
+        return response
+
 
 
 def setUp(test):
     zope.component.testing.setUp()
-    # Register Renderer Components
-    ztapi.provideUtility(IFactory, ReStructuredTextSourceFactory,
-                         'zope.source.rest')
-    ztapi.browserView(IReStructuredTextSource, '',
-                      ReStructuredTextToHTMLRenderer)
-    # Cheat and register the ReST renderer as the STX one as well.
-    ztapi.provideUtility(IFactory, ReStructuredTextSourceFactory,
-                         'zope.source.stx')
-    ztapi.browserView(IReStructuredTextSource, '',
-                      ReStructuredTextToHTMLRenderer)
-    setup.setUpTestAsModule(test, 'zope.app.apidoc.doctest')
-
+    xmlconfig.file('configure.zcml', zope.app.renderer)
+    zope.testing.module.setUp(test, 'zope.app.apidoc.doctest')
+    # Make RST the default for the sake of examples
+    test.globs['__docformat__'] = 'restructuredtext'
```

Comment:
The way we've been dropping `__docformat__` assignments from all zope.* packages, I think we should make ReST the default at the place that currently checks `__docformat__`.